### PR TITLE
AB#584 Add CLI install command

### PR DESCRIPTION
**Pull request by daniel-weisse**

### CODE REVIEW

**Review by m1ghtym0 — DISMISSED** [review]

**Review by thomasten — CHANGES_REQUESTED** [review]
Pls find out why tests are not running. I got the same problem locally when trying `go test ./...` or `go vet ./...`.

File: cli/cmd/install.go [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+		Args:  cobra.NoArgs,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			settings = cli.New()
+			hostname := domain
+			sim := simulation
+			noSgx := noSgxDevicePlugin
+			path := chartPath
+			clientPort := clientServerPort
+			meshPort := meshServerPort
+			return cliInstall(path, hostname, sim, noSgx, clientPort, meshPort, settings)
```
Comment (thomasten) [inline cli/cmd/install.go:46]:
```suggestion
			return cliInstall(chartPath, domain, simulation, noSgxDevicePlugin, clientServerPort, meshServerPort, settings)
```

File: cli/cmd/install.go [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+	cmd.Flags().IntVar(&meshServerPort, "mesh-server-port", 25554, "Set the mesh server port. Needs to be configured to the same port as in the data-plane marbles")
+	cmd.Flags().IntVar(&clientServerPort, "client-server-port", 25555, "Set the client server port. Needs to be configured to the same port as in your client tool stack")
+
+	return cmd
+}
+
+// cliInstall installs marblerun on the cluster
+func cliInstall(path string, hostname string, sim bool, noSgx bool, clientPort int, meshPort int, settings *cli.EnvSettings) error {
+	actionConfig := new(action.Configuration)
+	err := actionConfig.Init(settings.RESTClientGetter(), "marblerun", os.Getenv("HELM_DRIVER"), debug)
```
Comment (thomasten) [inline cli/cmd/install.go:64]:
```suggestion
	if err := actionConfig.Init(settings.RESTClientGetter(), "marblerun", os.Getenv("HELM_DRIVER"), debug); err != nil {
		return err
	}
```

File: cli/cmd/install.go [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+
+	// set overwrite values
+	var vals map[string]interface{}
+	if sim == true {
```
Comment (thomasten) [inline cli/cmd/install.go:68]:
```suggestion
	if sim {
```

File: cli/cmd/install.go [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+	// create helm installer
+	installer := action.NewInstall(actionConfig)
+	installer.CreateNamespace = true
+	installer.Namespace = "marblerun"
+	installer.ReleaseName = "marblerun-coordinator"
+
+	if path == "" {
+		// No chart was specified -> look for edgeless repository, if not present add it
+		err = getRepo("edgeless", "https://helm.edgeless.systems/stable", settings)
+		if err != nil {
```
Comment (thomasten) [inline cli/cmd/install.go:104]:
```suggestion
		if err := getRepo("edgeless", "https://helm.edgeless.systems/stable", settings); err != nil {
```

File: cli/cmd/install.go [hunk 5]
```diff
@@ -0,0 +1,193 @@
...
+			return err
+		}
+	}
+	chart, err := loader.Load(path)
+	if err != nil {
+		return err
+	}
+
+	err = chartutil.ValidateAgainstSchema(chart, vals)
+	if err != nil {
```
Comment (thomasten) [inline cli/cmd/install.go:118]:
```suggestion
	if err := chartutil.ValidateAgainstSchema(chart, vals); err != nil {
```

File: cli/cmd/install.go [hunk 6]
```diff
@@ -0,0 +1,193 @@
...
+		return err
+	}
+
+	_, err = installer.Run(chart, vals)
+	if err != nil {
```
Comment (thomasten) [inline cli/cmd/install.go:123]:
```suggestion
	if _, err := installer.Run(chart, vals); err != nil {
```

**Review by thomasten — CHANGES_REQUESTED** [review]
looks good, some style nitpicks

@m1ghtym0 @daniel-weisse Any idea how we could test this?

**Review by thomasten — APPROVED** [review]

**Pull request merged**